src/Compilers/CSharp/Portable/Compiler/TypeCompilationState.cs

```diff
@@ -164,6 +172,59 @@ public void Free()
             }
 
             _wrappers = null;
+            _constructorInitializers = null;
+        }
+
+        // Add an edge to the ctor-initializer graph
+        internal void NoteConstructorInitializer(MethodSymbol method1, MethodSymbol method2, CSharpSyntaxNode syntax)
+        {
+            if (_constructorInitializers == null)
+            {
+                _constructorInitializers = new Dictionary<MethodSymbol, Tuple<MethodSymbol, CSharpSyntaxNode>>();
```

Comment:
I would prefer to use another dictionary implementation whose enumeration is dependent on something like the location within a source file, or even the order in which items get added. The order of enumerating `Keys` determines which constructor in the chain is the target of the new error.